lib/rester/service/resource/params.rb

```diff
@@ -22,17 +24,34 @@ def strict?
         !!options[:strict]
       end
 
+      def frozen?
+        !!@_frozen
+      end
+
       def freeze
         @_validators.freeze
         @_required_fields.freeze
         @_defaults.freeze
         @_all_fields.freeze
+        @_frozen = true
       end
 
       def required_params
         @_required_fields.dup
       end
 
+      def defaults
+        @_defaults.dup
+      end
+
+      def all_fields
+        @_all_fields.dup
+      end
+
+      def validators
+        @_validators.dup
+      end
+
```

Comment:
Make these three methods `protected`.  Create a protected section above the `private` section with these in it.

`protected` means that the methods can only be called by other instances of the class.